@xen-orchestra/lite/src/components/ActionButton.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import LoadingButton, { LoadingButtonProps } from '@mui/lab/LoadingButton'
+import { withState } from 'reaclette'
+
+interface ParentState {}
+
+interface State {
+  isLoading: boolean
+}
+
+interface Props extends LoadingButtonProps {
+  onClick: (e: React.MouseEvent, data?: { [key: string]: unknown }) => Promise<void>
+  // to pass props with the following patern: "data-something"
+  [key: string]: unknown
+}
+
+interface ParentEffects {}
+
+interface Effects {
+  _onClick: React.MouseEventHandler<HTMLButtonElement>
+}
+
+interface Computed {
+  data: { [key: string]: unknown }
+}
+
+const ActionButton = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({ isLoading: false }),
+    effects: {
+      _onClick: function (e) {
+        this.state.isLoading = true
+        return this.props.onClick(e, this.state.data).finally(() => (this.state.isLoading = false))
+      },
+    },
+    computed: {
+      data: (_, props) => {
+        const _data: Record<string, unknown> = {}
+        Object.keys(props).forEach(key => {
```

Comment:
1. Does `Object.keys` trigger the props getters for Reaclette?
2. Even if it does, this wouldn't be recomputed for new data props
3. We don't need data props values for rendering

So maybe this should only be computed when the button is clicked, what do you think?